cmd/edit.go

```diff
@@ -0,0 +1,140 @@
+// Copyright © 2017 The Kubicorn Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"fmt"
+	"github.com/kris-nova/kubicorn/cutil/logger"
+	"github.com/kris-nova/kubicorn/state"
+	"github.com/kris-nova/kubicorn/state/fs"
+	"github.com/spf13/cobra"
+	"io/ioutil"
+	"os"
+	"os/exec"
+)
+
+type EditOptions struct {
+	Options
+	Editor string
+}
+
+var eo = &EditOptions{}
+
+var editCmd = &cobra.Command{
+	Use:   "edit <NAME>",
+	Short: "Edit a cluster state",
+	Long:  `Use this command to edit a state.`,
+	Run: func(cmd *cobra.Command, args []string) {
+		if len(args) == 0 {
+			ao.Name = strEnvDef("KUBICORN_NAME", "")
+		} else if len(args) > 1 {
+			logger.Critical("Too many arguments.")
+			os.Exit(1)
+		} else {
+			eo.Name = args[0]
+		}
+
+		err := RunEdit(eo)
+		if err != nil {
+			logger.Critical(err.Error())
+			os.Exit(1)
+		}
+
+	},
+}
+
+func init() {
+	editCmd.Flags().StringVarP(&eo.StateStore, "state-store", "s", strEnvDef("KUBICORN_STATE_STORE", "fs"), "The state store type to use for the cluster")
+	editCmd.Flags().StringVarP(&eo.StateStorePath, "state-store-path", "S", strEnvDef("KUBICORN_STATE_STORE_PATH", "./_state"), "The state store path to use")
+	editCmd.Flags().StringVarP(&eo.Editor, "editor", "e", strEnvDef("KUBICORN_DEFAULT_EDITOR", "vi"), "The editor used to edit the state store")
```

Comment:
Can we read from `$EDITOR` here? Most operating systems will already respect this value, so most users (myself included) will already have this set.. to `/usr/bin/emacs` in my case :stuck_out_tongue_winking_eye: